src/defrag.c

```diff
@@ -685,11 +685,11 @@ void defragKey(defragCtx *ctx, dictEntry *de) {
     newsds = activeDefragSds(keysds);
     if (newsds) {
         dictSetKey(db->dict[slot], de, newsds);
-        if (dbSize(db, DB_EXPIRES)) {
+        if (dictSize(db->expires[slot])) {
```

Comment:
i think this change is gonna clash with #12822 .
let's re-consider after it's merged (a day or two)